File: Spa-Salon-FrontEnd/src/types/service.ts
export interface Service {
  id: string;
  name: string;
  description?: string;
  durationMinutes: number;
  price: number;
  isActive: boolean;
  employeeId: string;
  employeeName: string;
}

export interface ServiceDetail {
  id: string;
  name: string;
  description?: string;
  durationMinutes: number;
  price: number;
  isActive: boolean;
  employeeId: string;
  employeeFirstName: string;
  employeeLastName: string;
  employeePosition: string;
}

export interface CreateService {
  name: string;
  description?: string;
  durationMinutes: number;
  price: number;
  isActive?: boolean;
  employeeId: string;
}

export interface UpdateService {
  name?: string;
  description?: string;
  durationMinutes?: number;
  price?: number;
  isActive?: boolean;
  employeeId?: string;
}

File: Spa-Salon-BackEnd/Domain/Entities/Service.cs
namespace Spa_Salon_BackEnd.Domain.Entities;

public class Service
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int DurationMinutes { get; set; }
    public bool IsActive { get; set; } = true;

    public Guid EmployeeId { get; set; }
    public Employee Employee { get; set; } = null!;

    public ICollection<AppointmentService> AppointmentServices { get; set; } = new List<AppointmentService>();
}
